v0/destinations/hs/transform.js

```diff
@@ -1,244 +1,609 @@
+        // can be due to network lag or processor being busy
+        const isDuplicate = identifyResponseList.find(data => {
+          return (
+            data.properties.email === ev.message.body.JSON.properties.email
+          );
+        });
+        if (!isEmpty(isDuplicate)) {
+          // array is being shallow copied hence changes are affecting the original reference
+          isDuplicate.properties = ev.message.body.JSON.properties;
+        } else {
+          // appending unique events
+          identifyResponseList.push({
+            properties: ev.message.body.JSON.properties
+          });
+        }
+        metadata.push(ev.metadata);
+      });
+    } else {
+      // update operation
+      chunk.forEach(ev => {
+        // eslint-disable-next-line no-param-reassign
+        ev.message.body.JSON.properties = getCRMUpdatedProps(
+          ev.message.body.JSON.properties
+        );
+        // update has id and properties
+        const id = ev.message.endpoint.split("/").pop();
+
+        // duplicate contactId is not allowed in batch
+        // updating the existing one to avoid duplicate
+        // as same event can fire in batch one of the reason
+        // can be due to network lag or processor being busy
+        const isDuplicate = identifyResponseList.find(data => {
+          return data.id === id;
+        });
+        if (!isEmpty(isDuplicate)) {
+          isDuplicate.properties = ev.message.body.JSON.properties;
+        } else {
+          // appending unique events
+          identifyResponseList.push({
+            id,
+            properties: ev.message.body.JSON.properties
+          });
+        }
+        metadata.push(ev.metadata);
+      });
+    }
+
+    batchEventResponse.batchedRequest.body.JSON = {
+      inputs: identifyResponseList
+    };
+
+    if (batchOperation === "create") {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_CREATE_NEW_CONTACT;
+    } else {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_UPDATE_NEW_CONTACT;
+    }
+    batchEventResponse.batchedRequest.headers = message.headers;
+    batchEventResponse.batchedRequest.params = message.params;
+
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata,
+      destination
+    };
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+};
+
+const batchEvents = destEvents => {
+  let batchedResponseList = [];
+  const trackResponseList = [];
+  let createContactEventsChunk = [];
+  let updateContactEventsChunk = [];
+  const arrayChunksIdentifyCreateContact = [];
+  const arrayChunksIdentifyUpdateContact = [];
+  destEvents.forEach((event, index) => {
+    // handler for track call
+    if (event.message.method === "GET") {
```

Comment:
track calls with new API will have endpoint `POST`, need to handle that, otherwise those will also go in, with identify.

Reply:
Done